clang/lib/Sema/SemaExprCXX.cpp

```diff
@@ -9063,7 +9063,8 @@ Sema::BuildExprRequirement(
     concepts::ExprRequirement::ReturnTypeRequirement ReturnTypeRequirement) {
   auto Status = concepts::ExprRequirement::SS_Satisfied;
   ConceptSpecializationExpr *SubstitutedConstraintExpr = nullptr;
-  if (E->isInstantiationDependent() || ReturnTypeRequirement.isDependent())
+  if (E->isInstantiationDependent() || E->getType()->isPlaceholderType() ||
+      ReturnTypeRequirement.isDependent())
```

Comment:
This makes sense in my head, as what is currently an unresolved overload expr(or other overload expr) could potentially become legal after instantiation I think?  So this should result in this being instantiated later/checked for correctness then.

Reply:
I've been trying to think about whether checking `UnresolvedLookupExpr` would make more sense but
I think what you have works in all cases.

Reply:
I was concerned about other expression types as well, like `OverloadExpr` or placeholders that got put in elsewise, so I was hoping this was sufficiently generic here.